project-base/storefront/components/Blocks/Product/Filter/DeferredFilterPanel.tsx

```diff
@@ -0,0 +1,15 @@
+import { FilterPanelProps } from './FilterPanel';
+import { SkeletonModuleFilterPanel } from 'components/Blocks/Skeleton/SkeletonModuleFilterPanel';
+import dynamic from 'next/dynamic';
+import { useDeferredRender } from 'utils/useDeferredRender';
+
+const FilterPanel = dynamic(() => import('./FilterPanel').then((component) => component.FilterPanel), {
+    ssr: false,
+    loading: () => <SkeletonModuleFilterPanel />,
```

Comment:
Hmm, I cannot wrap my head around this. What is this used for? Why some deferred components have this and some not? 🤔

Reply:
The logic is that if you can provide a simple skeleton with no props, you can set `ssr:false` and use it as the loading component. If it was not used, the change of components would be visible and would cause layout shift. This way it is fixed. 

In some places this is not provided, because we cannot provide it (placeholder accepts props). In such situations, we import both components on the server (so `ssr:true`) and we only switch between them based on the `shouldRender` boolean. 

I will add description of all cases to the documentation and cookbooks.